Fails to build with Clang 13 (due to -Werror)

Regressed by https://github.com/llvm/llvm-project/commit/abf3ca61e3235. Workaround is `-DBUILD_WERROR=OFF`.

```c++
$ export CC=clang13 CXX=clang++13
$ cmake -B /tmp/build -G Ninja
$ cmake --build /tmp/build
[...]
In file included from layers/core_validation.cpp:70:
In file included from layers/core_validation.h:31:
In file included from layers/image_layout_map.h:30:
layers/subresource_adapter.h:67:5: error: definition of implicit copy assignment operator for 'Subresource' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
    Subresource(const Subresource& from) = default;
    ^
layers/subresource_adapter.h:305:42: note: in implicit copy assignment operator for 'subresource_adapter::Subresource' first required here
        *static_cast<Subresource*>(this) = encoder_->Decode(index);
                                         ^
1 error generated.
```
http://www.ipv6proxy.net/go.php?u=http://gohan04.nyi.freebsd.org/data/mainamd64PR258209-default/2021-10-16_18h41m03s/logs/errors/vulkan-validation-layers-1.2.196.log

--- a/layers/subresource_adapter.h
+++ b/layers/subresource_adapter.h
@@ -70,6 +70,8 @@ struct Subresource : public VkImageSubresource {
         : VkImageSubresource({aspect_mask_, mip_level_, array_layer_}), aspect_index(aspect_index_) {}
     Subresource(VkImageAspectFlagBits aspect_, uint32_t mip_level_, uint32_t array_layer_, uint32_t aspect_index_)
         : Subresource(static_cast<VkImageAspectFlags>(aspect_), mip_level_, array_layer_, aspect_index_) {}
+
+    Subresource& operator=(const Subresource&) = default;
 };
 
 // Subresource is encoded in (from slowest varying to fastest)
@@ -415,6 +417,7 @@ class ImageRangeGenerator {
     inline const IndexRange& operator*() const { return pos_; }
     inline const IndexRange* operator->() const { return &pos_; }
     ImageRangeGenerator& operator++();
+    ImageRangeGenerator& operator=(const ImageRangeGenerator&) = default;
 
   private:
     bool Convert2DCompatibleTo3D();

--- a/layers/synchronization_validation.h
+++ b/layers/synchronization_validation.h
@@ -272,6 +272,7 @@ class ResourceAccessState : public SyncStageAccess {
         VkPipelineStageFlags2KHR exec_scope;
         SyncStageAccessFlags access_scope;
         OrderingBarrier() = default;
+        OrderingBarrier(const OrderingBarrier &) = default;
         OrderingBarrier(VkPipelineStageFlags2KHR es, SyncStageAccessFlags as) : exec_scope(es), access_scope(as) {}
         OrderingBarrier &operator=(const OrderingBarrier &) = default;
         OrderingBarrier &operator|=(const OrderingBarrier &rhs) {
@@ -733,6 +734,7 @@ class AccessContext {
     struct TrackBack {
         std::vector<SyncBarrier> barriers;
         const AccessContext *context;
+        TrackBack(const TrackBack &) = default;
         TrackBack(const AccessContext *context_, VkQueueFlags queue_flags_,
                   const std::vector<const VkSubpassDependency2 *> &subpass_dependencies_)
             : barriers(), context(context_) {

--- a/layers/vk_mem_alloc.h
+++ b/layers/vk_mem_alloc.h
@@ -3906,6 +3906,8 @@ class VmaStlAllocator
 public:
     const VkAllocationCallbacks* const m_pCallbacks;
     typedef T value_type;
+
+    VmaStlAllocator(const VmaStlAllocator&) = default;
     
     VmaStlAllocator(const VkAllocationCallbacks* pCallbacks) : m_pCallbacks(pCallbacks) { }
     template<typename U> VmaStlAllocator(const VmaStlAllocator<U>& src) : m_pCallbacks(src.m_pCallbacks) { }

